engine/crates/engine-v2/src/plan/planning/boundary.rs

```diff
@@ -0,0 +1,526 @@
+                ) {
+                    continue;
+                } else {
+                    return false;
+                }
+            }
+
+            // -- Plannable by the parent --
+            let field = self.schema.walker().walk(item.field_id);
+            let parent_logic = self
+                .maybe_parent
+                .expect("Cannot have requirements without a parent plan");
+            // no need to check for requires here, they're only relevant when it's a
+            // plan root field and this is a nested field. So we expect the data source
+            // to be able to provide anything it needed for a nested object it provides.
+            if parent_logic.plan_id() == parent_field_plan_id && parent_logic.is_providable(field.id()) {
+                if let Some(boundary_field) =
+                    self.try_planning_boundary_field(parent_logic, parent_selection_set_id, item)
+                {
+                    boundary_fields.insert(item.field_id, boundary_field);
+                    continue;
+                }
+            }
+
+            // -- Plannable by existing children --
+            for i in 0..self.children.len() {
+                let plan_id = self.children[i];
+                // ensures we don't have cycles between plans ensuring they can only depend on
+                // plan_ids lower than theirs. Could be better.
+                if plan_id <= parent_field_plan_id {
+                    continue;
+                }
+                let resolver_id = self.get_planned_resolver(plan_id).resolver_id;
+                for FieldResolverWalker {
+                    resolver,
+                    field_requires,
+                } in field.resolvers()
+                {
+                    if resolver.id() != resolver_id
+                        && self.could_plan_requirements_on_previous_plans(
+                            plan_id,
+                            boundary_fields,
+                            origin_bound_field_id,
+                            field_requires,
+                        )
+                    {
+                        let logic = &AttributionLogic::CompatibleResolver {
+                            plan_id,
+                            resolver,
+                            providable: field
+                                .provides_for(&resolver)
+                                .map(|field_set| field_set.into_owned())
+                                .unwrap_or_default(),
+                        };
+                        if let Some(boundary_field) =
+                            self.try_planning_boundary_field(logic, parent_selection_set_id, item)
+                        {
+                            boundary_fields.insert(item.field_id, boundary_field);
+                            continue 'requires;
+                        }
+                    }
+                }
+            }
+
+            // -- Add new child plan --
+            // eventually?
```

Comment:
This seems like it could be elaborated on (or just removed).  Don't think anyone but you knows what it means

Reply:
good point